packages/frontend/core/src/modules/workbench/entities/workbench.ts

```diff
@@ -13,17 +14,27 @@ interface WorkbenchOpenOptions {
 }
 
 export class Workbench extends Entity {
-  readonly views$ = new LiveData([
-    this.framework.createEntity(View, { id: nanoid() }),
-  ]);
+  constructor(private readonly defaultState: WorkbenchDefaultState) {
+    super();
+  }
+
+  readonly activeViewIndex$ = new LiveData(this.defaultState.activeViewIndex);
+  readonly basename$ = new LiveData(this.defaultState.basename);
+
+  readonly views$: LiveData<View[]> = new LiveData(
+    this.defaultState.views.map(meta => {
+      return this.framework.createEntity(View, {
+        id: meta.id,
+        defaultLocation: meta.path,
+      });
+    })
+  );
 
-  activeViewIndex$ = new LiveData(0);
   activeView$ = LiveData.computed(get => {
     const activeIndex = get(this.activeViewIndex$);
     const views = get(this.views$);
-    return views[activeIndex];
+    return views[activeIndex]; // todo: this could be null
```

Comment:
if it's hard to fix default active view index to 0, I perfer not persisting it, we can still get all tabs back though

Reply:
it is persisted to allow syncing the index conveniently across different views. I think it is ok in this case.